packages/contracts-bedrock/src/L2/interfaces/IOptimismSuperchainERC20.sol

```diff
@@ -2,38 +2,23 @@
 pragma solidity ^0.8.0;
 
 // Interfaces
-import { IERC20Solady } from "src/vendor/interfaces/IERC20Solady.sol";
-import { ISuperchainERC20Extensions, ISuperchainERC20Errors } from "src/L2/interfaces/ISuperchainERC20.sol";
+import { ISuperchainERC20 } from "src/L2/interfaces/ISuperchainERC20.sol";
 
-/// @title IOptimismSuperchainERC20Extension
+/// @title IOptimismSuperchainERC20
 /// @notice This interface is available on the OptimismSuperchainERC20 contract.
-///         We declare it as a separate interface so that it can be used in
-///         custom implementations of SuperchainERC20.
-interface IOptimismSuperchainERC20Extension is ISuperchainERC20Extensions, ISuperchainERC20Errors {
-    /// @notice Emitted whenever tokens are minted for an account.
-    /// @param account Address of the account tokens are being minted for.
-    /// @param amount  Amount of tokens minted.
-    event Mint(address indexed account, uint256 amount);
-
-    /// @notice Emitted whenever tokens are burned from an account.
-    /// @param account Address of the account tokens are being burned from.
-    /// @param amount  Amount of tokens burned.
-    event Burn(address indexed account, uint256 amount);
-
-    /// @notice Allows the L2StandardBridge to mint tokens.
-    /// @param _to     Address to mint tokens to.
-    /// @param _amount Amount of tokens to mint.
+interface IOptimismSuperchainERC20 is ISuperchainERC20 {
```

Comment:
It has specific errors of the abstract implementation `SuperchainERC20` that extends `ICrosschainERC20`. We extend it here so that the `OptimismSuperchainERC20` has the correct ABI.

Reply:
Hmm I thought I heard that SuperchainERC20 was being renamed to CrosschainERC20, and therefore I'd only expect to see `CrosschainERC20` throughout this PR. I may be out of the loop there though, so just want to better understand the relationship between the two

Reply:
Unresolved this since I still see both and want to make sure I understand the decision/architecture here

Reply:
The `SuperchainERC20` is a specific implementation of the `ICrosschainERC20` designed for the Optimism ecosystem. It’s called Superchain because it relies on the `SuperchainTokenBridge` for minting and burning, which is only necessary within the Superchain. Other L2 networks will need to implement the `ICrosschainERC20` in their own way.